lib/private/AppConfig.php

```diff
@@ -30,413 +33,1041 @@
+	 *
+	 * @return array<string, string> [configKey => configValue]
+	 * @since 29.0.0
+	 */
+	public function getAllValues(string $app, string $key = '', bool $filtered = false): array {
+		$this->assertParams($app, $key);
+		$this->loadConfig(null, $filtered); // if we want to filter values, we need to get sensitivity
+
+		$values = array_merge($this->fastCache[$app], $this->lazyCache[$app] ?? []);
+
+		if (!$filtered) {
+			return $values;
 		}
 
-		return [];
+		/**
+		 * Using the old (deprecated) list of sensitive values.
+		 */
+		foreach ($this->getSensitiveKeys($app) as $sensitiveKeyExp) {
+			$sensitiveKeys = preg_grep($sensitiveKeyExp, array_keys($values));
+			foreach ($sensitiveKeys as $sensitiveKey) {
+				$this->sensitive[$app][$sensitiveKey] = true;
+			}
+		}
+
+		$result = [];
+		foreach ($values as $key => $value) {
+			$result[$key] = ($this->sensitive[$app][$key] ?? false) ? IConfig::SENSITIVE_VALUE : $value;
+		}
+
+		return $result;
 	}
 
-	public function getSortedKeys($data) {
-		$keys = array_keys($data);
-		sort($keys);
-		return $keys;
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $key config key
+	 * @param bool $lazy lazy config
+	 *
+	 * @return array<string, string> [appId => configValue]
+	 * @since 29.0.0
+	 */
+	public function searchValues(string $key, bool $lazy = false): array {
+		$this->assertParams('', $key, true);
+		$this->loadConfig($lazy);
+		$values = [];
+		/** @var array<array-key, array<array-key, mixed>> $cache */
+		($lazy) ? $cache = &$this->lazyCache : $cache = &$this->fastCache;
+		foreach (array_keys($cache) as $app) {
+			if (isset($cache[$app][$key])) {
+				$values[$app] = $cache[$app][$key];
+			}
+		}
+
+		return $values;
 	}
 
+
 	/**
-	 * Gets the config value
+	 * @inheritDoc
 	 *
-	 * @param string $app app
-	 * @param string $key key
-	 * @param string $default = null, default value if the key does not exist
-	 * @return string the value or $default
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param string $default default value (optional)
+	 * @param string $lazy name of the lazy group (optional)
 	 *
-	 * This function gets a value from the appconfig table. If the key does
-	 * not exist the default value will be returned
+	 * @return string stored config value or $default if not set in database
+	 * @throws InvalidArgumentException
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueInt()
+	 * @see self::getValueFloat()
+	 * @see self::getValueBool()
+	 * @see self::getValueArray()
 	 */
-	public function getValue($app, $key, $default = null) {
-		$this->loadConfigValues();
+	public function getValueString(string $app, string $key, string $default = '', bool $lazy = false): string {
+		$this->assertParams($app, $key);
+		$this->loadConfig($lazy);
+		($lazy) ? $cache = &$this->lazyCache : $cache = &$this->fastCache;
+
+		/** @psalm-suppress UndefinedVariable */
```

Comment:
Instead of surpressing, fix the code and do it properly:
```suggestion
		if ($lazy) {
			$cache = &$this->lazyCache;
		} else {
			$cache = &$this->fastCache;
		}
```

Reply:
i'll do that, but I am more inclined to think that this is an issue with psalm, not my code :-]